daemon/graphdriver/driver.go

```diff
@@ -142,10 +142,36 @@ func GetDriver(name, home string, options []string) (Driver, error) {
 func New(root string, options []string) (driver Driver, err error) {
 	for _, name := range []string{os.Getenv("DOCKER_DRIVER"), DefaultDriver} {
 		if name != "" {
+			logrus.Infof("[graphdriver] trying provided driver %q", name) // so the logs show specified driver
 			return GetDriver(name, root, options)
 		}
 	}
 
+	// Guess for prior driver
```

Comment:
@dqminh for example, i've just tested that:
* clean slate
* starting a daemon with `-s devicemapper`
* `docker pull busybox`
* kill daemon
* starting daemon with `-s overlay` this time
* `docker pull busybox'

as expected, `overlay` directory is newer. Now:
* daemon with `-s overlay`
* `docker run busybox uptime`
* kill daemon
* start daemon with `-s devicemapper`
* `docker run busybox uptime`

now, my most recent used driver is `devicemapper`, but the newer mtime directory is still `overlay`.

I think deducing by cross referencing what directory is present and the priority list is a fairly solid guess. There is log output for the curious. Going beyond this would require additional state on disk, and may honestly be a very small percentage of real world cases.